core/templates/pages/exploration-editor-page/services/user-email-preferences.service.ts

```diff
@@ -16,77 +16,98 @@
-    var MESSAGE_TYPE_SUGGESTION = 'suggestion';
-    var MESSAGE_TYPE_FEEDBACK = 'feedback';
-    return {
-      init: function(
-          feedbackNotificationsMuted, suggestionNotificationsMuted) {
-        this.feedbackNotificationsMuted = feedbackNotificationsMuted;
-        this.suggestionNotificationsMuted = suggestionNotificationsMuted;
-      },
-      /**
-       * @return {boolean} Whether the feedback notification is muted.
-       */
-      areFeedbackNotificationsMuted: function() {
-        return this.feedbackNotificationsMuted;
-      },
-      /**
-       * @return {boolean} Whether the suggestion notification is muted.
-       */
-      areSuggestionNotificationsMuted: function() {
-        return this.suggestionNotificationsMuted;
-      },
-      /**
-       * Set the message type to feedback and mute to true or false.
-       * @param {boolean} mute - Whether the feedback notification is muted.
-       */
-      setFeedbackNotificationPreferences: function(mute) {
-        this.saveChangeToBackend({
-          message_type: MESSAGE_TYPE_FEEDBACK,
-          mute: mute
-        });
-      },
-      /**
-       * Set the message type to suggestion and mute to true or false.
-       * @param {boolean} mute - Whether the suggestion notification is muted.
-       */
-      setSuggestionNotificationPreferences: function(mute) {
-        this.saveChangeToBackend({
-          message_type: MESSAGE_TYPE_SUGGESTION,
-          mute: mute
-        });
-      },
-      /**
-       * Save the change of message_type and mute to backend.
-       * @param {object} requestParams - Info about message_type and mute.
-       */
-      saveChangeToBackend: function(requestParams) {
-        var that = this;
-        var emailPreferencesUrl = UrlInterpolationService.interpolateUrl(
-          '/createhandler/notificationpreferences/<exploration_id>', {
-            exploration_id: ExplorationDataService.explorationId
-          }
-        );
-        return $http.put(emailPreferencesUrl, requestParams).then(
-          function(response) {
-            var data = response.data;
-            AlertsService.clearWarnings();
-            that.init(
-              data.email_preferences.mute_feedback_notifications,
-              data.email_preferences.mute_suggestion_notifications);
-          }
-        );
-      }
-    };
+export interface EmailPreferencesData {
+  'email_preferences': {
+    'mute_feedback_notifications': boolean,
+    'mute_suggestion_notifications': boolean
   }
-]);
+}
+
+export interface RequestParams {
+  'message_type' : string,
+  mute: boolean
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserEmailPreferencesService {
+  MESSAGE_TYPE_SUGGESTION = 'suggestion';
+  MESSAGE_TYPE_FEEDBACK = 'feedback';
+  feedbackNotificationsMuted: boolean;
+  suggestionNotificationsMuted: boolean;
+  constructor(
+    private alertsService: AlertsService,
+    private backendApiService: UserEmailPreferencesBackendApiService
```

Comment:
rename this to `userEmailPreferencesBackendApiService`

Reply:
Done.